### ISSUE TITLE
No Method to Get Orbiter version in Lua

### ISSUE THREAD

**Issue by ThChicken** [orbitersim/orbiter#415]
There do not appear to be any equivalent methods to oapiGetOrbiterVersion () for Lua scripting. This prevents script developers from being able to check the Orbiter version and activate/deactivate methods according to the capabilities of the installed Orbiter version.

**Comment by schnepe2** [comment]
- Orbiter 2010: No, not at all
- Orbiter 2016: The function is `oapi.get_orbiter_version()` (although, you might need [the DLLs provided here](https://www.orbiter-forum.com/threads/orbiter-beta-lua-development.35459/))
- OpenOrbiter. Yes (again `oapi.get_orbiter_version()`) Example:
   ```
   %term.out(oapi.get_orbiter_version())
   240111
   ```

**Comment by ThChicken** [comment]
To be clear, does OpenOrbiter require those special .dlls for oapi.get_orbiter_version() or is that fully incorporated?

**Comment by schnepe2** [comment]
No, OpenOrbiter has it since day one! So no need for anything else. You can use it right away!

**Comment by ThChicken** [comment]
One more question - what is the behavior of this function if it is used when it is not available (Orbiter 2016 without the special .dlls for example)? Is there any way to avoid a crash in such an instance?

**Comment by schnepe2** [comment]
Ask a Lua specialist. It might be possible to check for methods at runtime and then decide to not call them.
Other than that Lua **is a scripted language** therefore you can always do silly things like `oapi.dumbMethodThatDoesNotExist()`.
And everyone would agree that this will crash!

**Comment by schnepe2** [comment]
A quick search found this:
https://stackoverflow.com/questions/25403979/lua-only-call-a-function-if-it-exists

**Comment by ThChicken** [comment]
I'll close this issue. Thanks for the information.
